skcosmo/preprocessing/KernelFlexibleCenterer.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+import sklearn as sk
+from sklearn.base import TransformerMixin, BaseEstimator
+
+
+class KernelFlexibleCenterer(TransformerMixin, BaseEstimator):
+    """Kernel centering method, similar to KernelCenterer,
+    but with additional parameters, relative to which centering
+    is carried out:
+    ref_cmean: an array with means for each column.
+    ref_rmean: an array with means for each row.
+    ref_mean: an average for the whole kernel matrix
+    reference: a reference, relative to which centering is carried out"""
+
+    def __init__(
+        self, *, reference=None, ref_cmean=None, ref_rmean=None, ref_mean=None
+    ):
+        """Initialize KernelFlexibleCenterer. Specifies the parameters relative to which the  kernel
+        matrix will be centered."""
+
+        self.reference = reference
+        self.ref_cmean = ref_cmean
+        self.ref_rmean = ref_rmean
+        self.ref_mean = ref_mean
+
+    def fit(self, K, y=None):
+        """Fit KernelFlexibleCenterer
+        Parameters
+        ----------
+        K : ndarray of shape (n_samples, n_samples)
+            Kernel matrix.
+        y : None
```

Comment:
then why do we need it?

Reply:
I don't know exactly. For some reason, everywhere in sklearn.preprocessing we pass y, although often it is simply ignored. I decided to do the same.